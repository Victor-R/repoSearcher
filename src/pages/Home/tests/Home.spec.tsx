import { render, screen, act, fireEvent } from "@testing-library/react";
import { Home } from "..";
import { BrowserRouter } from "react-router-dom";
import fetchMock from "jest-fetch-mock";
import { mockedUsersResponse } from "./mocks";

describe("Pages: Home", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    fetchMock.resetMocks();
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it("should match snapshot", () => {
    const { container } = render(<Home />, { wrapper: BrowserRouter });

    expect(container).toMatchSnapshot();
  });

  it("should search for users", async () => {
    fetchMock.mockResponse(JSON.stringify(mockedUsersResponse));

    render(<Home />, { wrapper: BrowserRouter });

    const searchInput = screen.getByPlaceholderText("Search for users...");

    fireEvent.change(searchInput, { target: { value: "Gandalf" } });

    act(() => {
      jest.runAllTimers();
    });

    expect(fetchMock).toBeCalled();
  });
});
