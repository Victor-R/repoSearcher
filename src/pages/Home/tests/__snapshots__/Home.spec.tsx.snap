// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages: Home should match snapshot 1`] = `
.c2 {
  display: flex;
  align-items: center;
  border: 1px solid #2b2a33;
  border-radius: 15px;
  height: 55px;
  background: #2b2a33;
  padding: 10px 20px;
}

.c2 svg {
  width: 26px;
  height: 26px;
  fill: white;
}

.c2:focus-within {
  border: 3px solid #763ead;
}

.c2:focus-within svg {
  fill: #763ead;
}

.c3 {
  flex: 1;
  background: none;
  border: none;
  font-size: 2em;
  color: white;
  outline: none;
}

.c0 {
  width: 100vw;
  height: 100vh;
  display: grid;
  grid-template-columns: 0.1fr 1fr;
  gap: 16px;
  background: rgb(2, 0, 36);
  background: linear-gradient(     180deg,     rgba(2, 0, 36, 1) 0%,     rgba(49, 8, 131, 1) 44%,     rgba(153, 90, 195, 1) 100%   );
  padding-top: 50px;
  overflow: hidden;
}

.c1 {
  display: flex;
  flex-direction: column;
  gap: 10px;
  padding-left: 10px;
}

.c4 {
  padding: 20px 8px;
  display: flex;
  flex-direction: column;
  gap: 10px;
  height: 90vh;
  overflow-y: auto;
}

.c4>svg {
  margin: 0 auto;
  width: 50px;
  height: 50px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <input
          class="c3"
          placeholder="Search for users..."
          value=""
        />
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </div>
    </div>
    <div
      class="c4"
    />
  </div>
</div>
`;
